pkg/fswatcher/fswatcher_test.go

```diff
@@ -22,113 +23,163 @@ func TestWatcher(t *testing.T) {
 	indirectSymlink := filepath.Join(tmp, "foo", "symlink", "nested")
 	targetFile := filepath.Join(tmp, "target")
 
+	cases := []struct {
+		name string
+		work func() // os level file operations
+		want []Event
+	}{
+		{
+			name: "create untracked dir",
+			work: func() {
+				// create $tmp/foo/ (this should not emit an event)
+				fooDirectory := filepath.Join(tmp, "foo")
+				err := os.MkdirAll(fooDirectory, 0777)
+				require.NoError(t, err)
+			},
+			want: []Event{},
+		},
+		{
+			name: "create and write file",
+			work: func() {
+				// create $tmp/file
+				err := os.WriteFile(regularFile, data, 0777)
+				require.NoError(t, err)
+			},
+			want: []Event{
+				{Name: regularFile, Op: Create | Write},
+			},
+		},
+		{
+			name: "update file",
+			work: func() {
+				// create $tmp/file
+				err := os.WriteFile(regularFile, []byte("some new data"), 0777)
+				require.NoError(t, err)
+			},
+			want: []Event{
+				{Name: regularFile, Op: Write},
+			},
+		},
+		{
+			name: "create symlink",
+			work: func() {
+				// symlink $tmp/symlink -> $tmp/target
+				err := os.WriteFile(targetFile, data, 0777)
+				require.NoError(t, err)
+				err = os.Symlink(targetFile, regularSymlink)
+				require.NoError(t, err)
+			},
+			want: []Event{
+				{Name: regularSymlink, Op: Create | Write},
+			},
+		},
+		{
+			name: "create nested file",
+			work: func() {
+				// create $tmp/foo/bar/nested
+				err := os.MkdirAll(filepath.Dir(nestedFile), 0777)
+				require.NoError(t, err)
+				err = os.WriteFile(nestedFile, data, 0777)
+				require.NoError(t, err)
+			},
+			want: []Event{
+				{Name: nestedFile, Op: Create | Write},
+			},
+		},
+		{
+			name: "create indirect symlink",
+			work: func() {
+				// symlink $tmp/foo/symlink -> $tmp/foo/bar (this will emit an event on indirectSymlink)
+				err := os.Symlink(nestedDir, directSymlink)
+				require.NoError(t, err)
+			},
+			want: []Event{
+				{Name: indirectSymlink, Op: Create | Write},
+			},
+		},
+		{
+			name: "redirect symlink",
+			work: func() {
+				// redirect $tmp/symlink -> $tmp/file (this will not emit an event)
+				err := os.Remove(regularSymlink)
+				require.NoError(t, err)
+				err = os.Symlink(regularFile, regularSymlink)
+				require.NoError(t, err)
+			},
+			want: []Event{},
+		},
```

Comment:
AFAIU we're "watching" `regularSymlink` and this test case change it from `→ $tmp/target` to `→ $tmp/file` right? If correct, don't we need to emit something here?

Reply:
That's a good question. I kept all the original test cases as is, so I didn't come up with this one.

Reply:
After fixing a bug, this test was now updated with an event for the `regularSymlink: UPDATE`